include/picongpu/plugins/output/WriteSpeciesCommon.hpp

```diff
@@ -145,16 +145,20 @@ struct FreeMemory
         if (ptr != nullptr)
         {
 #if( PMACC_CUDA_ENABLED == 1 )
-            auto rc = cudaFreeHost(ptr);
-            /* cupla can't handle foreign memory allocated with `cudaHostAlloc`
-             * therefore the cupla error `cuplaErrorMemoryAllocation` is ignored
-             */
-            if(rc != cuplaErrorMemoryAllocation)
-                CUDA_CHECK(rc)
+/* cupla is not supporting the function cudaHostAlloc to create mapped memory.
+ * Therefore we need to call the native CUDA function cudaFreeHost to free the memory.
+ * Due to the renaming of cuda functions with cupla via macros we need to remove
+ * the renaming to get access to the native cuda function.
+ * @todo this is a workaround plese fix me. We need to investigate if
+ * it is possible to have mapped/unified memory in alpaka.
+ */
+#   undef cudaFreeHost
```

Comment:
I would guard this with a `#ifdef`

Reply:
The reason why one would guard a `def` or `undef` is to make it more flexible for other hacks/workarounds that might (un)set.

You can keep it as is under the mentioned assumption, but that would make it more flexible.